### ISSUE TITLE
Rule Change: Enforce strict syntax for @charset

### ISSUE THREAD

**Issue by thecalamiity** [eslint/css#185]
### What rule do you want to change?

no-invalid-at-rules

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```css
@charset 'iso-8859-15'; /* Invalid: single quotes */
@charset  "UTF-8";      /* Invalid: two spaces */
```

### What does the rule currently do for this code?

Currently, the rule only checks if the prelude is a `<string>`, so both `@charset 'iso-8859-15';` and `@charset  "UTF-8";` are not reported as errors.

### What will the rule do after it's changed?

After the change, the rule will report an error for any `@charset` that:

- Does not use double quotes for the encoding,
- Does not have exactly one space after `@charset`,
- Is not immediately terminated with a semicolon.

### Participation

- [x] I am willing to submit a pull request to implement this change.
